### PR TITLE
Create UserService

### PR DESCRIPTION
**Pull request by allendomingo**
Create UserService class to replace UserCache.
Uses redis for caching as well

**Changelist**:
- [x] Modify User class constructor to accept IUser
- [x] Refactor/simplify how redisClient and other dependencies are passed to the services
- [x] Create User service that uses redis for caching

### CODE REVIEW

**Review by MKLythus — APPROVED** [review]

File: src/services/userService.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    /**
+     * Save game/s to user's game list to database and then to the cache
+     * @param id - id of the specified user
+     * @param gameNames - name of the games to be saved
+     */
+    saveToUserGames(id : string, gameNames : Array<string>) : Promise<User> {
+        return this.getUser(id).then((user) => {
+            user.addGameNames(gameNames);
+            return Users.findOneAndUpdate(
```
Comment (RayAdrian) [inline src/services/userService.ts:103]: @allendomingo thoughts about separating DB operations in separate folders and files `repositories` from services and controllers?
Makes it easier for changing DBs in the future if needed

Comment (allendomingo) [reply src/services/userService.ts:103]: Good suggestion. Would that mean the current services being made would then be controllers, and there would be a separate service for purely db transactions?
ie. Responsibilities split, with services handling pure db, and controllers doing caching, maybe validation as well

Comment (allendomingo) [reply src/services/userService.ts:103]: If we are doing this, I still suggest to do it in a follow up PR

Comment (RayAdrian) [reply src/services/userService.ts:103]: Yeap 💯 . Yes it'll be done in a separate PR

**Review by RayAdrian — APPROVED** [review]
🚀

**Pull request merged**